### DISCUSSION TITLE
Github App - search/code REST API not working for organization private repositories

### DISCUSSION THREAD

**Discussion by alvmarcos** [community/community#113651] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

Hello, I have been searching for an answer for quite a while but was not able to get any relevant information or instructions. So far, the issue is the following:
- Have been successfully using Github Personal Token to search code in organization private repositories via API. Now moving to a more automated process where we want to use Github App installation tokens to perform the actions.
- I created a Github App, I transferred ownership of it to an organization I belong to. The app was installed in the organization with access to all repositories. I confirmed it was installed in private repositories by using [this](https://docs.github.com/en/rest/apps/installations?apiVersion=2022-11-28#list-repositories-accessible-to-the-app-installation) endpoint.
My use case is to identify specific files with certain content/characteristics in such private repos (using an app installation token). Unfortunately, [this](https://docs.github.com/en/rest/search/search?apiVersion=2022-11-28#search-code) api always return an empty response (with 200 status code) when I use the qualifier repo:ORG/PRIVATE_REPO_NAME + optional search terms (e.g. name or contents of the file or even nothing else). According to the documentation of the endpoint (linked above) it should work with App installation tokens, without any additional permissions. I have confirmed this endpoint works with public repositories (returns a non-empty response). 
I have also confirmed the search/repositories endpoint works (is able to list in the response) private repositories of the organization. 
I have also used the token with Graphql API and it works for repository searches, although this API is not able to provide code search.

Thus, I believe this is either a bug or an expected behaviour that is not correctly documented. I am looking forward for any feedback on permissions the app may need or the way to make this work. Thank you in advance. Kind regards.

**Comment by amber-beasley-liatrio** [comment]
This could be a related issue: https://github.com/orgs/community/discussions/48102

**Comment by shayki5** [comment]
I facing the exact issue!
Search API working with regular PAT in **private** repos but doesn't work with GitHub App authentication!!

**Comment by vz443** [comment]
Is there any workaround/solution for this, other than using personal pat
